linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2860,17 +2895,32 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
           }
 
         case Closure(arrow, captureParams, params, body, captureValues) =>
-          val innerFunction = {
-            desugarToFunctionInternal(arrow, params, body, isStat = false,
-                Env.empty(AnyType).withParams(captureParams ++ params))
-          }
+          val capturesBuilder = List.newBuilder[(js.ParamDef, js.Tree)]
+          val replacementsBuilder = Map.newBuilder[LocalName, js.Tree]
 
-          val captures = for {
+          for {
             (param, value) <- captureParams.zip(captureValues)
           } yield {
-            (transformParamDef(param), transformExpr(value, param.ptpe))
+            val newValue = transformExpr(value, param.ptpe)
+
+            if (!param.mutable && mayElideIIFECapture(value, allowThis = arrow)) {
+              replacementsBuilder += param.name.name -> newValue
+            } else {
+              capturesBuilder += transformParamDef(param) -> newValue
```

Comment:
In fact we could do better here. If we're not in a loop, we can introduce local synthetic vars in the current function, that are initialized with the capture values, and use them as replacement in the function. This way, when we're not in a loop, we can always get rid of the IIFE.

Reply:
Is that better for JITs? IIUC in terms of code size it is marginal (see below), but it would significantly complicate the Emitter (because we need the cooperation of the unnesting infrastructure).

```
((<name>) => (<body>))(<value>) // 12 bytes overhead
// vs
var <name> = <value>;
<body> // 9 bytes overhead
```

Reply:
Well, no IIFE is always better than an IIFE, so if it allows to get rid of the IIFE altogether, it's better for the JIT.

That said, it would indeed complicate everything because it requires unnesting, which I had not thought about. Since this case shouldn't really happen in practice (scalac only generates `Closure`s where the capture values are `val` references or `this`, and the optimizer only replaces those with other `val` references or `this` or constants), it's not worth adding support for this.